src/index.ts

```diff
@@ -857,6 +867,93 @@ class Spanner extends GrpcService {
     });
   }
 
+  getInstanceConfig(name: string): Promise<GetInstanceConfigResponse>;
+  getInstanceConfig(name: string, callback: GetInstanceConfigCallback): void;
+  /**
+   * Gets the instance configuration with the specified name.
+   */
+  /**
+   * @typedef {array} GetInstanceConfigResponse
+   * @property {object[]} 0 The metadata of the instance config.
+   * @property {string} 0.name The unique identifier for the instance config.
+   * @property {string} 0.displayName The name of the instance config as it
+   *     appears in UIs.
+   * @property {google.spanner.admin.instance.v1.IReplicaInfo[]} 0.replicas The replicas used by
+   *     this instance config.
+   * @property {string[]} 0.leaderOptions The possible leader options for this instance config.
+   */
+  /**
+   * @callback GetInstanceConfigCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} instanceConfig The metadata of the instance config.
+   * @param {string} instanceConfig.name The unique identifier for the instance
+   *     config.
+   * @param {string} instanceConfig.displayName The name of the instance config
+   *     as it appears in UIs.
+   * @param {google.spanner.admin.instance.v1.IReplicaInfo[]} instanceConfig.replicas The replicas used by
+   *     this instance config.
+   * @param {string[]} 0.leaderOptions The possible leader options for this instance config.
+   */
+  /**
+   * Get a specific instance config.
+   *
+   * Wrapper around {@link v1.InstanceAdminClient#getInstanceConfig}.
+   *
+   * @see {@link v1.InstanceAdminClient#getInstanceConfig}
+   * @see [GetInstanceConfig API Documentation](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.admin.instance.v1#google.spanner.admin.instance.v1.InstanceAdmin.GetInstanceConfig)
+   *
+   * @param {string} [name] The name of the instance config to get.
+   * @param {GetInstanceConfigCallback} [callback] Callback function.
+   * @returns {Promise<GetInstanceConfigResponse>}
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   *
+   * spanner.getInstanceConfig('nam6', function(err, instanceConfig) {
+   *   // `instanceConfig` is an instance configuration descriptor.
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * spanner.getInstanceConfig().then(function(data) {
+   *   const instanceConfig = data[0];
+   * });
+   */
+  getInstanceConfig(
+    name: string,
+    optionsOrCallback?: CallOptions | GetInstanceConfigCallback,
```

Comment:
I might be overthinking it here but should we future-proof the options by making a new interface for `GetInstanceConfigOptions` that includes `gaxOptions`? I feel like recently, we've introduced lots of new options into existing RPCs that made it very tricky to support in Node.js

Reply:
That's true. I've updated the function definition to use a specific `GetInstanceConfigOptions` that includes gax options.